autosklearn/automl.py

```diff
@@ -440,6 +440,18 @@ def fit(
             else:
                 per_run_time_limit = self._per_run_time_limit
 
+            # Make sure that at least 2 models are created for the ensemble process
+            min_ensemble_time = time_left_for_smac // 10
+            num_models = time_left_for_smac // per_run_time_limit
+            if num_models < 2:
+                per_run_time_limit = time_left_for_smac//2 - min_ensemble_time
```

Comment:
The thinking behind min_ensemble_time is that you need to make sure that the ensemble builder has enough time to fit/evaluate at least an ensemble of 2. So it will take smac  **time_left_for_smac//2**  seconds to build one model. Then, and additional **time_left_for_smac//2** to fit the second one.

So a total of time_left_for_smac has passed. However, time_left_for_ensembles will also be due (because time_left_for_smac is almost the same as time_left_for_ensembles), so no time for ensemble building in the worst-case scenario.

does it make sense?

Reply:
Okay, so that's some slack so that the ensemble can pick up the 2nd model? If that's the case, I don't think that's necessary because making sure that a 2nd model is started training is already giving us this slack (and you could therefore remove it).